sdk/component/panic.go

```diff
@@ -0,0 +1,39 @@
+package component
+
+import (
+	"context"
+	"runtime/debug"
+)
+
+type (
+	// PanicHandler defines a generic contract for handling panics following the recover semantics.
+	PanicHandler interface {
+		// HandlePanic handles a panic and returns an optional error with a signal on whether it should be
+		// fatal or not.
+		HandlePanic(ctx context.Context, err any) (error, bool)
+	}
+
+	defaultPanicHandler struct{}
+)
+
+// NewDefaultPanicHandler returns a new default panic handler.
+func NewDefaultPanicHandler() (*defaultPanicHandler, error) {
+	return &defaultPanicHandler{}, nil
+}
+
+// HandlePanic logs a panic and tells the runner to exit from the application.
+func (dph *defaultPanicHandler) HandlePanic(ctx context.Context, err any) (error, bool) {
```

Comment:
Very simple default panic handler which returns an error and tells the app to shutdown.

This can be overridden for example, to use sentry to report problems.